### ISSUE TITLE
Tracking tokens/clones is currently tedious

### ISSUE THREAD

**Issue by Fenrirthviti** [Cockatrice/Cockatrice#4184]
**Similar Requests**
N/A

**Issue Description**
Currently, tokens have no way of informing the UI/player that they are tokens, aside from artwork. This is a big pain point, especially in EDH format where the board state can get pretty complicated. This gets especially complicated with clones, with the following example:

![img](https://pub.rachni.com/img/cockatrice_2020-11-24_03-33-10.png)

One of the cards here is a token copy that is a clone of the card. There is no way of telling which one is which, and can lead to annoyance/confusion when needing to, say, destroy one. One will disappear when destroyed, but the other goes to the graveyard, and knowing which needs to be chosen can be important to avoid having to roll back the board state if you accidentally choose the wrong one. We can use annotations as a workaround, but that becomes fairly tedious.

There should be a clear UI paradigm that indicates a clone is a token (maybe a colored border of some kind, or just adding `(Token)` to the card name). @tooomm suggested that at one point it was discussed to add a `(T)` icon to the card, which would also work here as an indication. Personally, I think a colored border to indicate a non-standard card type might be the best choice for maximum "at a glance" board state information.

Tracking copies of the same token by incrementing a number at the end of their name would be sufficient for most use cases I can think of.

**Additional Context**
None at this time.

**Comment by tooomm** [comment]
+1 on a visual hint on tokens!

<br>

>Tracking copies of the same token by incrementing a number at the end of their name would be sufficient for most use cases I can think of.

You mean showing one token as e.g. `Warrior (4)` instead having 4 individual `Warrior` token?

Automatically merging tokens can lead to other problems and confusions with attacking and blocking etc.
IMO this doesn't need to change and user can use counters for combining tokens of the same kind already, same as e.g. a dice in paper.

**Comment by Fenrirthviti** [comment]
No, not merging tokens (though that's a separate suggestion for how to handle larger piles...). It would be:

Warrior (1)
Warrior (2)
Warrior (3)

Though the more I think about this, the less important I feel tracking which specific token is which is not very relevant, and won't come up often enough to really care too much about.

**Comment by n21lv** [comment]
**Proposed solution**
The easiest way to solve this is by adding an annotation saying "token" onto tokens and cloned objects. This is what I usually do anyway, since I have a few decks that care about whether an object is a token or not. This is the easiest solution to implement, as tokens can already be automatically annotated by the game.

**Acceptance criteria**
1. Whenever a player creates a token or clones an existing object, the resulting object SHOULD have an annotation saying "TOKEN".
2. The annotation MUST ignore the "Annotate card text on tokens" preference.
###### NB: The key words "MUST", "MUST NOT", "SHOULD" and "SHOULD NOT" are to be interpreted as described in IETF RFC 2119.

**Comment by ebbit1q** [comment]
currently cloning a card clones that card's annotation, would this TOKEN annotation replace it or be added at the end? should this text be configurable?

**Comment by n21lv** [comment]
I've been thinking about possible scenarios, and there is no way you can create a copy of a card that isn't a token. There are emblems, though, and things like Monarch and City's Blessing reminder cards, which are neither tokens nor emblems. So there are little reasons for making the word configurable.

The word "TOKEN" text should be put first, and any existing text should be appended to it. Otherwise, with a long enough annotation (or when default behaviour of annotating the tokens is enabled), it will be clipped and this will defy the purpose of adding it there in the first place.

**Comment by Fenrirthviti** [comment]
Annotations seems like a worse idea than visual or universal language, such
as icons, colors, and numbers. Text would then have to be translated and
is, frankly, just more clutter on an already visually dense game.

**Comment by n21lv** [comment]
> Annotations seems like a worse idea than visual or universal language, such as icons, colors, and numbers. Text would then have to be translated and is, frankly, just more clutter on an already visually dense game.

I agree with you, but making this game less cluttered while maintaining the possibility to see the boards of all players at the same time is a very difficult task. I will try to explain why annotating the token is probably the best solution we can hope for. 

Icon is not the best choice for a visual cue here, because icons will have to be scaled up or down depending on the size of the game window. Will it look good with 2 players? With 3..4? With 5..6? What if someone has A LOT of objects on their board? And if we also take card scaling into consideration (icons must not cover the card too much), it starts to add quite a lot to implementation complexity. How big should be the maximum area of a card this icon can cover? What icon to use so it is understandable enough for players? How to make sure that it is distinguishable enough, regardless of the card image? 

Numbers can also be difficult to notice, unless you mean using numbers as image overlay, but then it's the same as with icons. 

Coloured borders are another issue, because there are already two versions of the border colour (selected cards have red, cards that have "Toggle normal untapping" checked have purple), and they don't always work well together (try selecting a normal card and a card that has "Toggle normal untapping" checked at the same time, for example). Another issue is that it's hard to choose a colour that "works" regardless of the background. This is actually already an issue with the red colour used for the border of selected cards. I have a slight colour vision deficiency, so it's a bit of a bigger issue for me, but red on green/blue background isn't the best colour pair in terms of contrast _**anyway**_ -- check https://webaim.org/resources/contrastchecker/.

There is also another issue -- any added complexity also makes the whole project more bug-prone. Annotating the tokens with the word "TOKEN", therefore, is probably the best solution, because:

* it solves the stated issue (token copies are indistinguishable from "real" cards) in a way that's effective enough
* it uses existing functionality 
* introduces the smallest of possible changes
* which in turn introduces the smallest risk of breaking any existing functionality 

It will also be much easier to maintain. Don't forget that devs are doing all of this in their free time and without compensation, so any fancy-looking solutions, unfortunately, are really expensive in terms of effort.
